### PR TITLE
Bump `logback` version to 1.2.13+

### PR DESCRIPTION
**Pull request by jladieu**
Maintenance ticket: [AJ-1485](https://broadworkbench.atlassian.net/browse/AJ-1485)

For dependabot findings:
* https://github.com/DataBiosphere/terra-workspace-data-service/security/dependabot/46
* https://github.com/DataBiosphere/terra-workspace-data-service/security/dependabot/47

[AJ-1485]: https://broadworkbench.atlassian.net/browse/AJ-1485?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jladieu** [comment]
Version confirmation:
```
./gradlew service:dependencyInsight --configuration compileClasspath --dependency logback-classic --no-build-cache

> Configure project :service
The 'sonarqube' task depends on compile tasks. This behavior is now deprecated and will be removed in version 5.x. To avoid implicit compilation, set property 'sonar.gradle.skipCompile' to 'true' and make sure your project is compiled, before analysis has started.
The 'sonar' task depends on compile tasks. This behavior is now deprecated and will be removed in version 5.x. To avoid implicit compilation, set property 'sonar.gradle.skipCompile' to 'true' and make sure your project is compiled, before analysis has started.

> Task :service:dependencyInsight
ch.qos.logback:logback-classic:1.3.12
  Variant compile:
    | Attribute Name                 | Provided | Requested    |
    |--------------------------------|----------|--------------|
    | org.gradle.status              | release  |              |
    | org.gradle.category            | library  | library      |
    | org.gradle.libraryelements     | jar      | classes      |
    | org.gradle.usage               | java-api | java-api     |
    | org.gradle.dependency.bundling |          | external     |
    | org.gradle.jvm.environment     |          | standard-jvm |
    | org.gradle.jvm.version         |          | 17           |
   Selection reasons:
      - Was requested: didn't match versions 1.4.13, 1.4.12, 1.4.11, 1.4.10, 1.4.9, 1.4.8, 1.4.7, 1.4.6, 1.4.5, 1.4.4, 1.4.3, 1.4.2, 1.4.1, 1.4.0, 1.3.13
      - Selected by rule

ch.qos.logback:logback-classic:1.2.12 -> 1.3.12
\--- org.springframework.boot:spring-boot-starter-logging:2.7.18
     \--- org.springframework.boot:spring-boot-starter:2.7.18
          +--- compileClasspath (requested org.springframework.boot:spring-boot-starter)
          +--- org.springframework.boot:spring-boot-starter-batch:2.7.18
          |    \--- compileClasspath (requested org.springframework.boot:spring-boot-starter-batch)
          +--- org.springframework.boot:spring-boot-starter-jdbc:2.7.18
          |    +--- compileClasspath (requested org.springframework.boot:spring-boot-starter-jdbc)
          |    \--- org.springframework.boot:spring-boot-starter-batch:2.7.18 (*)
          +--- org.springframework.boot:spring-boot-starter-web:2.7.18
          |    \--- compileClasspath (requested org.springframework.boot:spring-boot-starter-web)
          +--- org.springframework.boot:spring-boot-starter-cache:2.7.18
          |    \--- compileClasspath (requested org.springframework.boot:spring-boot-starter-cache)
          +--- org.springframework.boot:spring-boot-starter-quartz:2.7.18
          |    \--- compileClasspath (requested org.springframework.boot:spring-boot-starter-quartz)
          +--- org.springframework.boot:spring-boot-starter-actuator:2.7.18
          |    \--- compileClasspath (requested org.springframework.boot:spring-boot-starter-actuator)
          \--- org.springframework.boot:spring-boot-starter-json:2.7.18
               \--- org.springframework.boot:spring-boot-starter-web:2.7.18 (*)

(*) - Indicates repeated occurrences of a transitive dependency subtree. Gradle expands transitive dependency subtrees only once per project; repeat occurrences only display the root of the subtree, followed by this annotation.

A web-based, searchable dependency report is available by adding the --scan option.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.4/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 619ms
1 actionable task: 1 executed
```

File: build.gradle [hunk 1]
```diff
@@ -40,6 +40,7 @@ subprojects {
         dependencies {
             dependency 'org.yaml:snakeyaml:2.0'
             dependency 'org.webjars:webjars-locator-core:0.53'
+            dependency 'ch.qos.logback:logback-classic:1.3.12+' // CVE-2023-6378
```
[the remark this answers is not in the document]
Comment (davidangb) [reply build.gradle:43]: https://docs.gradle.org/current/userguide/dependency_constraints.html for doc

[the remark this answers is not in the document]
Comment (jladieu) [reply build.gradle]: I like this approach better, but this is what I tried first and `dependencyManagement` completely ignored it which is how I ended up with what I pushed.

What do you think about specifying in both places belt & suspenders style?  I think the `implementation` style is a nice way of documenting the why, even if it doesn't have the effect we want.

**Review by nawatts — APPROVED** [review]

**Review by yuliadub — APPROVED** [review]
the approach you went with makes sense to me!

**Pull request merged**